kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/CheckpointInstance.java

```diff
@@ -36,19 +58,32 @@ public class CheckpointInstance
     public final long version;
     public final Optional<Integer> numParts;
 
+    public final CheckpointFormat format;
+
+    public final Optional<Path> filePath;
+
     public CheckpointInstance(String path) {
         String[] pathParts = getPathName(path).split("\\.");
```

Comment:
Add a check in the beginning to throw an error if the given path is not a checkpoint file name.

Preconditions.checkArgument(FileNames.isCheckpointFile(name), "not a valid checkpoint file name")

With the above you can remove `pathParts[1].equals("checkpoint")` from all three conditions below